interpreter/CMakeLists.txt

```diff
@@ -81,21 +81,17 @@ set(LLVM_ABI_BREAKING_CHECKS FORCE_OFF CACHE BOOL "" FORCE)
 
 set(CMAKE_REQUIRED_QUIET 1)  # Make the configuration of LLVM quiet
 
-if(ROOT_ARCHITECTURE MATCHES linuxarm64)
+if(ROOT_ARCHITECTURE MATCHES arm64)
   set(ROOT_CLING_TARGET "AArch64")
-elseif(ROOT_ARCHITECTURE MATCHES linuxarm)
+elseif(ROOT_ARCHITECTURE MATCHES arm)
   set(ROOT_CLING_TARGET "ARM")
-elseif(ROOT_ARCHITECTURE MATCHES linuxppc64gcc)
+elseif(ROOT_ARCHITECTURE MATCHES ppc64)
   set(ROOT_CLING_TARGET "PowerPC")
-elseif(ROOT_ARCHITECTURE MATCHES linuxs390)
+elseif(ROOT_ARCHITECTURE MATCHES s390)
   set(ROOT_CLING_TARGET "SystemZ")
-elseif(ROOT_ARCHITECTURE MATCHES linuxriscv64)
+elseif(ROOT_ARCHITECTURE MATCHES riscv64)
   set(ROOT_CLING_TARGET "RISCV")
-elseif(ROOT_ARCHITECTURE MATCHES linux)
-  set(ROOT_CLING_TARGET "X86")
-elseif(ROOT_ARCHITECTURE MATCHES macosxarm64)
-  set(ROOT_CLING_TARGET "AArch64")
-elseif(ROOT_ARCHITECTURE MATCHES macosx)
+elseif(ROOT_ARCHITECTURE MATCHES freebsd OR ROOT_ARCHITECTURE MATCHES linux OR ROOT_ARCHITECTURE MATCHES macosx)
```

Comment:
Why is `macosxarm64` no longer needed?

Reply:
It's taken care of by `ROOT_ARCHITECTURE MATCHES arm64`, which relieves us from the pain to specify every single platform cross architecture combination.

Reply:
That would be good but when does it change to `arm64`?  I see https://github.com/hahnjo/root/blob/interpreter-targets/cmake/modules/SetUpMacOS.cmake which sets it to the longer name.